### ISSUE TITLE
List workflow executions APIs should not require namespace and filter fields

### ISSUE THREAD

**Issue by mfateev** [temporalio/temporal#636]
## Expected Behavior
List workflows should default namespace to "default".

## Actual Behavior
Namespace argument is required.

**Comment by mfateev** [comment]
Also it requires the latest time if the earliest is set:

```java
    ListOpenWorkflowExecutionsRequest request =
        ListOpenWorkflowExecutionsRequest.newBuilder()
            .setNamespace("default")
            .setStartTimeFilter(
                StartTimeFilter.newBuilder()
                    .setEarliestTime(
                        TimeUnit.MILLISECONDS.toNanos(System.currentTimeMillis() - 100000))
                    //                    .setLatestTime(
                    //
                    // TimeUnit.MILLISECONDS.toNanos(System.currentTimeMillis() + 10000))
                    .build())
            .setTypeFilter(WorkflowTypeFilter.newBuilder().setName("GreetingWorkflow"))
            .build();
```
Fails with:
```
INVALID_ARGUMENT: EarliestTime in StartTimeFilter should not be larger than LatestTime.
```

**Comment by alexshtin** [comment]
Server doesn't have a concept of "default namespace". Clients are supposed to set default namespace if it is not set. This is what [GoSDK](https://github.com/temporalio/sdk-go/blob/6580cbe0aa41a8b515791f95c2c15bb37db1dab1/internal/internal_workflow_client.go#L706) does and `tctl` uses GoSDK for list operations. I would prefer Java and other SDK to do the same.

Dates validator seems to be fixed in #641 and #1631.
